bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/ModelServerRoutingV2.java

```diff
@@ -0,0 +1,99 @@
+/********************************************************************************
+ * Copyright (c) 2019-2022 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ ********************************************************************************/
+package org.eclipse.emfcloud.modelserver.emf.common;
+
+import static io.javalin.apibuilder.ApiBuilder.patch;
+import static io.javalin.apibuilder.ApiBuilder.post;
+import static io.javalin.apibuilder.ApiBuilder.put;
+import static io.javalin.apibuilder.ApiBuilder.ws;
+import static org.eclipse.emfcloud.modelserver.common.ModelServerPathParametersV2.MODEL_URI;
+import static org.eclipse.emfcloud.modelserver.emf.common.util.ContextResponse.error;
+import static org.eclipse.emfcloud.modelserver.emf.common.util.ContextResponse.missingParameter;
+
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.emfcloud.modelserver.common.ModelServerPathsV2;
+
+import com.google.inject.Inject;
+
+import io.javalin.Javalin;
+import io.javalin.http.Context;
+import io.javalin.websocket.WsConfig;
+import io.javalin.websocket.WsConnectContext;
+
+public class ModelServerRoutingV2 extends AbstractModelServerRouting {
+   protected static final Logger LOG = LogManager.getLogger(ModelServerRoutingV2.class);
+
+   protected final TransactionController transactionController;
+
+   @Inject
+   public ModelServerRoutingV2(final Javalin javalin, final ModelResourceManager resourceManager,
+      final ModelController modelController, final SchemaController schemaController,
+      final ServerController serverController, final SessionController sessionController,
+      final TransactionController transactionController) {
+
+      super(javalin, resourceManager, modelController, schemaController, serverController, sessionController);
+
+      this.transactionController = transactionController;
+   }
+
+   @Override
+   protected String getBasePath() { return ModelServerPathsV2.BASE_PATH; }
+
+   @Override
+   protected void apiEndpoints() {
+      super.apiEndpoints();
+
+      put(ModelServerPathsV2.MODEL_BASE_PATH, this::setModel); // Was PATCH in V1
+
+      patch(ModelServerPathsV2.MODEL_BASE_PATH, this::executeCommand); // Was PATCH /edit in V1
+
+      post(ModelServerPathsV2.TRANSACTION, this::createTransaction);
```

Comment:
This seems like a whole new feature. Is there any documentation when and how it is used? I also did not see any dedicated support for this in the ModelServerClient on the Javascript side.

Reply:
It's used by a new project: https://github.com/eclipse-emfcloud/modelserver-node

We decided to exclude it from the Javascript/Java client API at the moment; it's considered an implementation detail, to be used through the `modelserver-node` APIs (At the moment. But maybe it could be properly opened in the future).

Reply:
Yes, it's intended to be an internal API used by a coöperating network of model servers only, not by client applications. Perhaps that's a concept that could be formalized in the _Client API_ if we want to include the transaction endpoint in it? But we do not foresee other such internal APIs and as mentioned it isn't certain that this will not be opened to all clients when we've a clear idea of how that should work.

Reply:
I see, thank you for the explanation. I think removing the entries from the default paths to not explicitly expose it was the right thing to do for now. Thank you for that!